core/services/fluxmonitor/flux_monitor.go

```diff
@@ -102,6 +104,11 @@ func New(
 	}
 }
 
+// SetLogger sets and reconfigures the logger for the flux monitor service
+func (fm *concreteFluxMonitor) SetLogger(logger *logger.Logger) {
+	fm.logger = logger
```

Comment:
I think you're going to want to mutex this (and the one in the HeadTracker) given that `fm.logger` can be concurrently accessed (imagine a scenario where the request to set the logger arrives simultaneously with FM trying to use the logger -- could panic the node).

You'll also want a `.logger()` method (that also uses the mutex) any time `fm.logger` is accessed here in the FM code. Maybe rename the struct field to `log` or something

Reply:
Roger, makes sense to lock down the re assignment to the logger, something simple like: 

`
// SetLogger sets and reconfigures the logger for the flux monitor service
func (fm *concreteFluxMonitor) SetLogger(logger *logger.Logger) {
        fm.m.Lock()
	fm.logger = logger
        fm.m.Unlock()
}
`

and add the `sync.Mutex` type on the flux monitor struct